mongodb-modifications.md

```diff
@@ -0,0 +1,12 @@
+# MongoDB Modifications
+
+This file details the modifications made to nearley as compared to the upstream.
+
+## Differences
+
+tbd
+
+## Version History
+
+- v1.0.0: Corresponds to nearley v2.16.0
```

Comment:
I'm not super confident we'll diligently maintain this file. It's definitely worth having now, but if it gets out of date too easily, we should consider removing it.

Reply:
Agreed. I don't know how else to keep track of which version corresponds to upstream but hopefully we don't need to change the fork or pull upstream very often.